shared/src/dispatchers/ses/sendBulkTemplatedEmail.test.js

```diff
@@ -72,4 +72,158 @@ describe('sendBulkTemplatedEmail', () => {
     expect(applicationContext.logger.error.mock.calls.length).toEqual(1);
     expect(applicationContext.logger.info).toHaveBeenCalledTimes(1);
   });
+
+  it('should retry a failed mailing', async () => {
+    applicationContext
+      .getEmailClient()
+      .sendBulkTemplatedEmail.mockReturnValueOnce({
+        promise: () =>
+          Promise.resolve({
+            ResponseMetadata: {
+              RequestId:
+                '01000176a9ec8d81-a80255bb-8ab4-4049-ba1f-6abd5b7a8098-000000',
+            },
+            Status: [
+              {
+                MessageId:
+                  '01000176a9ec8d81-a80255bb-8ab4-4049-ba1f-6abd5b7a8098-000000',
+                Status: 'Success',
+              },
+              {
+                Error: 'Unknown error occurred. Please try again later',
+                Status: 'Failed',
+              },
+              {
+                Error: 'Unknown error occurred. Please try again later',
+                Status: 'Failed',
+              },
+            ],
+          }),
+      });
+
+    await sendBulkTemplatedEmail({
+      applicationContext,
+      destinations: [
+        {
+          email: 'test.email@example.com',
+          templateData: {
+            name: 'Guy Fieri',
+            welcomeMessage: 'Welcome to Flavortown',
+            whoAmI: 'The Sauce Boss',
+          },
+        },
+        {
+          email: 'test.email2@example.com',
+          templateData: {
+            name: 'Guy Fieri',
+            welcomeMessage: 'Welcome to Flavortown',
+            whoAmI: 'The Sauce Boss',
+          },
+        },
+        {
+          email: 'test.email3@example.com',
+          templateData: {
+            name: 'Guy Fieri',
+            welcomeMessage: 'Welcome to Flavortown',
+            whoAmI: 'The Sauce Boss',
+          },
+        },
+      ],
+      templateName: 'case_served',
+    });
+    expect(
+      applicationContext.getEmailClient().sendBulkTemplatedEmail.mock
+        .calls[1][0],
+    ).toMatchObject({
+      Destinations: [
+        {
+          Destination: {
+            ToAddresses: ['test.email2@example.com'],
+          },
+          ReplacementTemplateData: JSON.stringify({
+            name: 'Guy Fieri',
+            welcomeMessage: 'Welcome to Flavortown',
+            whoAmI: 'The Sauce Boss',
+          }),
+        },
+        {
+          Destination: {
+            ToAddresses: ['test.email3@example.com'],
+          },
+          ReplacementTemplateData: JSON.stringify({
+            name: 'Guy Fieri',
+            welcomeMessage: 'Welcome to Flavortown',
+            whoAmI: 'The Sauce Boss',
+          }),
+        },
+      ],
+    });
+    expect(
+      applicationContext.getEmailClient().sendBulkTemplatedEmail,
+    ).toHaveBeenCalledTimes(2);
+  });
+
+  it('should retry a failed mailing 10 times, and then throw an error', async () => {
```

Comment:
Should this be "7 times" ? - Also, what do you think about storing the `MAX_SES_RETRIES` on applicationContext, so we can mock it out / pull it from constants?

Reply:
thanks! it had been 10. I reduced it after adding the exponential backoff due to the anticipation of timing out a Lambda if un-sendable